services/raycast/installs.service.js

```diff
@@ -0,0 +1,68 @@
+import Joi from 'joi'
+import { metric } from '../text-formatters.js'
+import { anyInteger } from '../validators.js'
+import { BaseJsonService, NotFound, pathParams } from '../index.js'
+
+const schema = Joi.object({
+  download_count: anyInteger,
+}).required()
+
+export default class RaycastInstalls extends BaseJsonService {
+  static category = 'platform-support'
+
+  static route = {
+    base: 'raycast/installs',
+    pattern: ':user/:extension',
+  }
+
+  static openApi = {
+    '/raycast/installs/{user}/{extension}': {
+      get: {
+        summary: 'Raycast extension downloads count',
+        parameters: pathParams(
+          { name: 'user', example: 'Fatpandac' },
+          { name: 'extension', example: 'bilibili' },
+        ),
+      },
+    },
+  }
+
+  static _cacheLength = 7200
+
+  static render({ user, extension, downloadCount }) {
+    return {
+      label: 'Installs',
+      message: metric(downloadCount),
+      color: 'green',
+      link: [`https://www.raycast.com/${user}/${extension}`],
+    }
+  }
+
+  async fetch({ user, extension }) {
+    return this._requestJson({
+      schema,
+      url: `https://www.raycast.com/frontend_api/users/${user}/extensions/${extension}`,
```

Comment:
Hi. I haven't had a proper look at this code, but this looks like an internal API endpoint, rather than a public one.

Can you link to any documentation for this?

Reply:
Yes, this is not an open API. I found it on the [website](https://www.raycast.com/Fatpandac/bilibili) of Raycast and it does not have any documentation.

Reply:
OK thanks. The reason we say

> Badges should not obtain data from undocumented or reverse-engineered API endpoints.

in [our guidelines](https://github.com/badges/shields/blob/master/CONTRIBUTING.md#badge-guidelines)

is because we had some situations where a contributor worked out that a project had an API by inspecting the frontend code for their website or whatever, added a badge that calls it, and then we ended up relying on an API that wasn't really intended for public use. This then either annoyed the maintainers of the company/project behind it or we had no reasonable expectation of stability/compatibility over time (because devs maintaining the the API were making the assumption the only thing consuming it was their own frontend code) and stuff kept breaking.

That's looks like the situation we've got here so I am going to close this PR, but we could revisit in future if there is a reliable public source for the data. Other than the data source, it looks like a badge showing downloads for a raycast extension would be a reasonable badge to add.

Reply:
👋 I'm a Software Engineer @ Raycast. We don't have public API documentation available, but if you use `https://www.raycast.com/api/v1/extensions/petr/slack-status` instead of the `frontend_api` path, we're good with publishing the badge. We'll make sure the endpoint keeps being supported.

Reply:
Thanks for your reply 🙏, I will change to use this API soon.